src/python/pants/option/global_options.py

```diff
@@ -1268,6 +1268,24 @@ def register_bootstrap_options(cls, register):
             "may not be enabled.",
         )
 
+        register(
+            "--use-deprecated-python-macros",
```

Comment:
Since this flag will land as un-deprecated, it's starting out misnamed. Maybe invert to `--use-new-python-macros`?

Reply:
Eh I don't think it matters much. But in my head, it makes sense to see this in `pants.toml`:

```toml
[GLOBAL]
use_deprecated_python_macros = false
```

It makes clear that you're opting out of deprecated functionality. Technically, the macros aren't yet formally deprecated. But in spirit, they are deprecated. It's only a logistical reason why we can't formally deprecate them more quickly.

Reply:
Sounds good. What does the alternative look like..

```toml
[GLOBAL]
use_new_python_macros = true
```

Makes it clear that you're opting in to use the new macros before that is the default thing.

Hmm.. nah, that option name is more wrong, the new thing is not macros at all..